src/v/storage/key_offset_map.h

```diff
@@ -16,50 +16,54 @@
 
 namespace storage {
 
-// Map containing the latest offsets of each key.
+/**
+ * Map containing the latest offsets of each key.
+ */
 class key_offset_map {
 public:
-    // Returns true if the put was successful.
-    virtual bool put(compaction_key key, model::offset o) = 0;
+    key_offset_map() = default;
+    key_offset_map(const key_offset_map&) = delete;
+    key_offset_map& operator=(const key_offset_map&) = delete;
+    key_offset_map(key_offset_map&&) noexcept = default;
+    key_offset_map& operator=(key_offset_map&&) noexcept = default;
+    virtual ~key_offset_map() = default;
 
-    // Returns the latest offset for the given key put into the map.
+    /**
+     * Associate \p offset with the given \p key. If \p key already exists then
+     * the new mapping with override the existing mapping.
+     */
```

Comment:
nit: if the key already exists, the new mapping only overrides the existing mapping if it has a higher offset

Reply:
thanks fixed